pkg/scheduler/partition.go

```diff
@@ -262,6 +262,8 @@ func (pc *PartitionContext) isStopped() bool {
 // Handle the state event for the partition.
 // The state machine handles the locking.
 func (pc *PartitionContext) handlePartitionEvent(event objects.ObjectEvent) error {
+	pc.Lock()
+	defer pc.Unlock()
```

Comment:
Can we move this to line 269?
I think we do not need to lock the partition when calling `pc.stateMachine.Event(event.String(), pc.Name)`